### PR TITLE
fix: check npipe when executing podman compose on wsl vm

### LINKED ISSUE #20373: Podman compose on Windows
Following a customer call, I was trying to setup a compose sample.

I ended up with: Error: socket of machine is not set

Seems on Windows PodmanSocket is not returned by PodmanPipe

I used the sample from https://github.com/docker/awesome-compose/tree/master/nginx-golang-postgres

See
https://github.com/containers/podman/blob/dabf6c83a7793a2e5c76a54d5a41e6f4d2e62660/cmd/podman/compose.go#L187

### PR DESCRIPTION
**Pull request by lstocchi**
Closes #20373 

Add a check on the npipe if podman compose is run against a wsl vm and return the npipe path

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```
fix podman compose command not working with wsl
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@n1hility @ashley-cui PTAL

**Comment by rhatdan** [comment]
@lstocchi Please repush now that you set the [no new tests needed] flag

**Review by ashley-cui — COMMENTED** [review]
LGTM

**Comment by mheon** [comment]
Restarted three tests - looked like network flakes?

File: cmd/podman/compose.go [hunk 1]
```diff
@@ -184,12 +188,18 @@ func composeDockerHost() (string, error) {
 		if err != nil {
 			return "", fmt.Errorf("inspecting machine: %w", err)
 		}
-		if info.ConnectionInfo.PodmanSocket == nil {
-			return "", errors.New("socket of machine is not set")
-		}
 		if info.State != machine.Running {
 			return "", fmt.Errorf("machine %s is not running but in state %s", item.Name, info.State)
 		}
+		if item.VMType == "wsl" {
```
Comment (TomSweeneyRedHat) [inline cmd/podman/compose.go:194]: It would be nice to get some constants set up for the VMType at some point.

Comment (baude) [inline cmd/podman/compose.go:194]:
```
if machineProvider.VMType() == machine.WSLVirt
```

might be stronger type comparison ... though I readily admit I am not sure if that causes cyclical dependency issues or not.

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: cmd/podman/compose.go [hunk 2]
```diff
@@ -40,6 +40,10 @@ If you want to change the default behavior or have a custom installation path fo
 	Annotations:        map[string]string{registry.ParentNSRequired: ""}, // don't join user NS for SSH to work correctly
 }
 
+const (
+	pipePrefix = "npipe:////./pipe/"
```
Comment (baude) [inline cmd/podman/compose.go:44]: given this is only used once, should it just be inline or is having the const defined worth it.

**Comment by baude** [comment]
couple things for consideration ... i want @vrothberg to look at this before it merges.  @vrothberg if you are happy with it, feel free to merge without my comments being addressed

this makes me wonder about system connections and its future as well as whether this should be part of the provider interface so its hidden from cmdline stuff?

**Comment by baude** [comment]
/hold 

until @vrothberg  reviews

**Comment by vrothberg** [comment]
> this makes me wonder about system connections and its future as well as whether this should be part of the provider interface so its hidden from cmdline stuff?

I don't understand in detail what you mean but keeping connections around is needed for the non-machine remote case.

**Review by vrothberg — COMMENTED** [review]
@lstocchi can you address @baude's and @TomSweeneyRedHat's comments?  Then we can merge.

**Comment by lstocchi** [comment]
@vrothberg done

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

**Pull request merged**